torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -590,9 +593,21 @@ XLAGraphExecutor::ExecuteComputationWithBarrier(
     torch::lazy::BackendDataPtr handle =
         WrapXlaData(xla::ComputationClient::Get()->CreateDataPlaceholder(
             device.toString(), std::move(shape)));
+    // if SPMD is enabled, we assume all output will be replicated
+    if (ShardingUtil::UseVirtualDevice()) {
```

Comment:
Why we now start adding this for the dynamo path? We don't need this for the LTC path?

Reply:
Looks like this patch is dynamo exclusive... Should we hint this somewhere?

Reply:
the lazy code path already have this logic, in fact I copt this logic from lazy code path lol

Reply:
I smile an opportunity to merge two code paths more. But let's do it in a follow up.